frontend/src/component/ui/Table.spec.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Table } from './Table';
+
+describe('Table component', () => {
+  const sampleHeaders = ['Header1', 'Header2', 'Header3'];
+  const sampleRows = [
+    ['row1-cell1', 'row1-cell2', 'row1-cell3'],
+    ['row2-cell1', 'row2-cell2', 'row2-cell3'],
+    ['row3-cell1', 'row3-cell2', 'row3-cell3']
+  ];
```

Comment:
I recommend moving this into individual test case. You may feel this is creating repetition. However, it significantly easier for people who read the test to understand to understand the test case.

The data is the `arrange` step of the test: https://github.com/short-d/short#unit-testing